### PR TITLE
bzl: make integration tests actually work

### PR DESCRIPTION
**Pull request by mikedanese**
#59283
```release-note
NONE
```

### CODE REVIEW

**Comment by ixdy** [comment]
seems like we need to update to bazel 0.10 first.

also, we'll need changes to the prow job configs to switch to `--config=unit`, as otherwise we'll stop using the race detector in the interim.

File: build/root/Makefile [hunk 1]
```diff
@@ -544,7 +544,7 @@ else
 # We don't want to build any of the release artifacts when running tests.
 # Some things in vendor don't build due to empty target lists for cross-platform rules.
 bazel-test:
-	bazel test --build_tag_filters=-e2e,-integration --test_tag_filters=-e2e,-integration --flaky_test_attempts=3 -- \
+	bazel test --config=unit_test -- \
```
Comment (ixdy) [inline build/root/Makefile]: I think this should be `unit`, not `unit_test`?

**Comment by mikedanese** [comment]
@ixdy PTAL

**Comment by BenTheElder** [comment]
changes LGTM, leaving approval to @ixdy

**Comment by ixdy** [comment]
we probably should pair this with a change to the job configs in k/t-i?

**Comment by BenTheElder** [comment]
Yeah, flipping this over is somewhat problematic, we rely on some of these options being set, we'll need to make the master branch job(s) reference these config(s)

**Pull request merged**